src/libstd/sys/cloudabi/mod.rs

```diff
@@ -61,7 +59,7 @@ pub use libc::strlen;
 
 pub fn hashmap_random_keys() -> (u64, u64) {
     unsafe {
-        let mut v = mem::uninitialized();
+        let mut v = mem::MaybeUninit::uninit().assume_init();
```

Comment:
I feel like a type annotation is definitely warranted here, and probably should've been here before too.

Reply:
Yeah, great point. Will update.

Reply:
You are just inlining `mem::uninitialized` here. This does nothing to solve all the problems that `mem::uninitialized` as.

Please read that `assume_init` docs. You may only call that method *after* initialization has completed!

Reply:
Makes complete sense. I seem to have misunderstood the ask of the original issue and thought it was simply to replace the deprecated call. I will go back to square one with that in mind and actually fix the problem.